kusto/ingest/internal/filesystem/filesystem.go

```diff
@@ -55,44 +55,42 @@ type Ingestion struct {
 	uploadBlobFile   uploadBlobFile
 	transferManager  azblob.TransferManager
 
-	uploadBlockSize   int
-	uploadConcurrency uint16
+	bufferSize int
+	maxBuffers int
 }
 
 // Option is an optional argument to New().
 type Option func(s *Ingestion)
 
-// WithUploadBlockSize sets the block size for uploading blobs to kusto. Defaults to 8MB.
-func WithUploadBlockSize(size int) Option {
+// WithStaticBuffer sets a static buffer with a buffer size and max amount of buffers for uploading blobs to kusto.
+func WithStaticBuffer(bufferSize int, maxBuffers int) Option {
 	return func(s *Ingestion) {
-		s.uploadBlockSize = size
-	}
-}
-
-// WithUploadConcurrency sets the concurrency for uploading blobs to kusto. Defaults to 50.
-func WithUploadConcurrency(concurrency uint16) Option {
-	return func(s *Ingestion) {
-		s.uploadConcurrency = concurrency
+		s.bufferSize = bufferSize
+		s.maxBuffers = maxBuffers
 	}
 }
 
 // New is the constructor for Ingestion.
 func New(db, table string, mgr *resources.Manager, options ...Option) (*Ingestion, error) {
 	i := &Ingestion{
-		db:                db,
-		table:             table,
-		mgr:               mgr,
-		uploadBlobStream:  azblob.UploadStreamToBlockBlob,
-		uploadBlobFile:    azblob.UploadFileToBlockBlob,
-		uploadBlockSize:   BlockSize,
-		uploadConcurrency: Concurrency,
+		db:               db,
+		table:            table,
+		mgr:              mgr,
+		uploadBlobStream: azblob.UploadStreamToBlockBlob,
+		uploadBlobFile:   azblob.UploadFileToBlockBlob,
 	}
 
 	for _, opt := range options {
 		opt(i)
 	}
 
-	transferManager, err := azblob.NewSyncPool(i.uploadBlockSize, int(i.uploadConcurrency))
+	var transferManager azblob.TransferManager
+	var err error
+	if i.bufferSize == 0 {
```

Comment:
Greater then, also check the other value is a non zero positive

Reply:
Added clearer errors, as we discussed